### PR TITLE
Test resize postMessage is sent

### PR DESCRIPTION
**Pull request by zhouyx**
#4256

### CODE REVIEW

File: extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js [hunk 1]
```diff
@@ -210,14 +210,52 @@ function tests(name) {
           resizable: '',
         }, 'https://schema.org').then(element => {
           const impl = element.implementation_;
-          const attemptChangeSizeSpy = sandbox.spy(impl, 'attemptChangeSize');
-          impl.apiHandler_.updateSize_(217, 114);
-          expect(attemptChangeSizeSpy.callCount).to.equal(1);
-          expect(attemptChangeSizeSpy.firstCall.args[0]).to.equal(217);
-          expect(attemptChangeSizeSpy.firstCall.args[1]).to.equal(114);
+          const attemptChangeSize = sandbox.stub(impl, 'attemptChangeSize',
```
Comment (lannka) [inline extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js]: Don't quite understand what this test is trying to do here.
The name suggests it's to test a fallback is called when resize has overflow. But it's unclear what is the fallback.

Also, I saw the most logic is around the internal member field: `apiHandler_`. If the logic we're testing is inside `apiHandler_`, should we just move the test to `apiHandler_`'s unit test?

Comment (zhouyx) [inline extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js]: The description is confusing indeed. I changed it.

`ampHandler_` don't have unit test now. All tests are in `test-amp-ad-3p-impl` I agree we should move some tests to `ampHandler` but that involves more tests other than these two. I would prefer to do it in another PR.

**Comment by zhouyx** [comment]
@lannka PTAL

File: extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js (hunk 1 above)
Comment (lannka) [inline extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js]: I suggest we do the right thing in this PR. You don't need to move legacy tests over to the new file, but it's a good time to create such a separate unit test file, and have the new test in it. Otherwise, it's a waste of time to review and merge this PR in.

Meanwhile, it's better we test the external facing API of the `amp-ad-api-handler.js` but not internal private method. In this case, we should `window.postMessage('embed-size')`, instead of calling `updateSize_()`

**Comment by jridgewell** [comment]
Ping.

**Comment by zhouyx** [comment]
The test should be moved to `test-amp-ad-api-handler` instead. working on it

**Comment by zhouyx** [comment]
PTAL

**Comment by jridgewell** [comment]
LGTM

**Comment by zhouyx** [comment]
@lannka  Can I merge this to proceed with #4691

File: extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js [hunk 2]
```diff
@@ -74,5 +74,38 @@ describe('amp-ad-api-handler', () => {
...
+      return iframe.expectMessageFromParent('amp-' + JSON.stringify({
+        requestedWidth: 114,
+        requestedHeight: 217,
+        type: 'embed-size-denied',
+        sentinel: 'amp3ptest' + testIndex,
+      }));
+    });
+
+    it('should be able to use embed-size API, change size succeed', () => {
+      sandbox.stub(apiHandler.baseInstance_, 'attemptChangeSize', () => {
```
Comment (lannka) [inline extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js]: better to stub `adImpl` instead of accessing the private member field.

File: extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js [hunk 3]
```diff
@@ -74,5 +74,38 @@ describe('amp-ad-api-handler', () => {
         expect(iframe.style.visibility).to.equal('');
       });
     });
+
+    it('should be able to use embed-size API, change size deny', () => {
+      iframe.postMessageToParent({
```
Comment (lannka) [inline extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js]: to make the test case clearer, let's stub and reject the promise

File: extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js (hunk 2 above)
Comment (zhouyx) [inline extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js]: fixed

File: extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js (hunk 3 above)
Comment (zhouyx) [inline extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js]: fixed

**Comment by lannka** [comment]
LGTM
